rosbag2_transport/CMakeLists.txt

```
@@ -65,6 +69,12 @@ target_link_libraries(${PROJECT_NAME}
   yaml-cpp
 )
 
+rclcpp_components_register_node(
+  ${PROJECT_NAME} PLUGIN "rosbag2_transport::Player" EXECUTABLE Rosbag2Player)
```

```suggestion
  ${PROJECT_NAME} PLUGIN "rosbag2_transport::Player" EXECUTABLE player)
```
@MichaelOrlov one last thing... I believe it's better to rename the nodes, so that:
```
ros2 run rosbag2_transport player
```

The "Rosbag2" prefix is redundant IMHO.
And I believe that for node names lowercase convention is preferred.

(Same for Recorder here below)

---

@roncapat Out of curiosity, is it possible to make it `ros2 run rosbag2 player` without renaming `rosbag2_transport` to the `rosbag2`?
Maybe using an alias for the entry point?

---

Or maybe if can get rid of `rosbag2_transport` keyword in front. To be `ros2 run Rosbag2Player` ?

---

I believe that the semantic of the `ros2 run` command is 'package_name node_name' and can't be changed. We can only act on the node name.

Maybe we can then find a way to register the node under package "rosbag2". But it may be very tricky.

However, in the long term (I can open an Issue for later implementation) I just had a potentially nice idea: to upgrade `ros2 bag play` CLI support with a parameter for an existing container name. This way, the CLI would load the component in an existing container, start playing (and eventually unloading it at the end of the playback). This would be a neat feature IMHO. Thoughts?

Summing up: simplest solution here is at least for now just to choose a simple and idiomatic node name. We may act on moving the nodes to `rosbag2` package later on.

---

(To register the nodes under "rosbag2" instead of "rosbag2_transport" we may need to hack the inner workings of this: https://github.com/ros2/rclcpp/blob/rolling/rclcpp_components%2Fcmake%2Frclcpp_components_register_node.cmake)

---

@roncapat Ahh never mind. Feel free to go ahead and commit your suggestions with signoff.

---

As regards:
> This way, the CLI would load the component in an existing container, start playing (and eventually unloading it at the end of the playback). This would be a neat feature IMHO. Thoughts?

Curious why it would be useful at all if we already have entry point for `ros2 bag play` and `ros2 bag record`?
I only envision a case when we can run player or recorder and specify settings yaml file instead of the CLI parameters directly. Something like `ros2 bag play --settings playback_settings.yaml`

The same questions apply to how to use `ros2 run rosbag2_transport player` in a real use case? How we will provide settings or node parameters in this case?

---

> As regards:
> > This way, the CLI would load the component in an existing container, start playing (and eventually unloading it at the end of the playback). This would be a neat feature IMHO. Thoughts?
> 
> Curious why it would be useful at all if we already have entry point for `ros2 bag play` and `ros2 bag record`?

Imagine people have already scripted their `ros2 bag play/record` commands (for example, in some launch files). With a small modification (like the addition of one CLI parameter) they will be able to "upgrade their setup" and inject player/recorder in their component containers. Increased performance with almost zero effort (if you have specified all parameters in the CLI command, you won't have to migrate to a .yaml configuration file).


> The same questions apply to how to use `ros2 run rosbag2_transport player` in a real use case? How we will provide settings or node parameters in this case?

Like normal node parameters via YAML file. Notice that this also enables people to use `Node()` objects in `LaunchConfiguration()` of launchfiles instead of the current only way to run player/recorder via launchfiles (to specify the full CLI command).

In the next days I'll write some docs to push as separate PR. It may be the right context to sum up all the past, present and potentially future ways to run players/recorders if it's ok for you :)

---

@roncapat As regards:
> magine people have already scripted their ros2 bag play/record commands (for example, in some launch files). With a small modification (like the addition of one CLI parameter) they will be able to "upgrade their setup" and inject player/recorder in their component containers. Increased performance with almost zero effort (if you have specified all parameters in the CLI command, you won't have to migrate to a .yaml configuration file).

I don't really understand about what increased performance and benefits you are talking about there. Maybe with documentation or some real-world examples will be clear to me.

As regards:
> The same questions apply to how to use `ros2 run rosbag2_transport player` in a real use case? How we will provide settings or node parameters in this case?

 > >   Like normal node parameters via YAML file. Notice that this also enables people to use Node() objects in LaunchConfiguration() of launchfiles instead of the current only way to run player/recorder via launchfiles (to specify the full CLI command).

Still unclear to me how node parameters will be settled with `ros2 run rosbag2_transport player` command. What you described seems more relevant for launch files launching composed nodes via the launcher. But it is irrelevant to the `ros2 run rosbag2_transport player` command as far as understand.

> In the next days I'll write some docs to push as separate PR. It may be the right context to sum up all the past, present and potentially future ways to run players/recorders if it's ok for you :)

Anyway, documentation would be very helpful and grateful to have.